tldr/semantic.py

```
@@ -12,19 +12,26 @@
 """
 
+import contextlib
 import json
 import logging
 import os
 import sys
 from concurrent.futures import ProcessPoolExecutor, as_completed
 from dataclasses import dataclass, field
 from pathlib import Path
-from typing import List, Optional, Tuple, Dict, Any
+from typing import Iterator, List, Optional, Tuple, Dict, Any
```

Ruff (0.15.10)

[warning] Import from `collections.abc` instead: `Iterator`

Import from `collections.abc`

(UP035)